Feature `mezzio/template` v3 compatibility

ghostwriter opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | yes
| QA            | yes

### Description

This pull request introduces compatibility improvements and code quality enhancements for the Mezzio Plates renderer. 

* Updated the `mezzio/mezzio-template` dependency in `composer.json` to support both v2 and v3.

* Added a the deprecated `Mezzio\Template\TemplatePath` class in `resources/Mezzio/Template/TemplatePath.php` to maintain upgrade compatibility until the next major release `mezzio/mezzio-platesrenderer` v3.

* Added the `Mezzio\Template` namespace in the autoloader configuration in `composer.json` and included the `resources` directory for code checks in `phpcs.xml.dist`.

* Added the `#[Override]` attribute to methods in `EscaperExtension`, `UrlExtension`, and `PlatesRenderer` to explicitly mark method overrides.

ghostwriter committed:
Add #[Override] attribute to overridden methods

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Add resources directory to PHPCS checks

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update dependencies and autoload paths

Expanded mezzio/mezzio-template version constraint to include ^3.0 and added autoloading for Mezzio\Template namespace.

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Add TemplatePath class for compatibility

Introduces a the deprecated TemplatePath class to maintain upgrade compatibility with mezzio/template v3.0.0. This class will be removed in mezzio/mezzio-platesrenderer v3.0.0.

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

gsteel reviewed (changes requested):
Thanks @ghostwriter 👍 

Would you mind addressing the Psalm issues?

Adding `/** @return list<TemplatePath> */` to `$renderer->getPaths()` should do it.

Noted that the BC Breaks are the result of `ExceptionInterface` now inheriting from `Throwable` in `mezzio-template@2.x`

ghostwriter committed:
Update getPaths docblock with return type

Replaces the inherited docblock for getPaths with a more specific return type annotation, clarifying that it returns a list of TemplatePath objects.

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update psalm-baseline.xml

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

gsteel reviewed (approved):
Thanks @ghostwriter :)
